models/sipnet/R/read.restart.SIPNET.R

```diff
@@ -1,29 +1,53 @@
-read.restart.SIPNET <- function(outdir,run.id,time,IC,prior){
+##' @title read.restart.SIPNET
+##' @name  read.restart.SIPNET
+##' @author Ann Raiho \email{araiho@@nd.edu}
+##' 
+##' @param outdir      output directory
+##' @param runid       run ID
+##' @param time        year that is being read
+##' @param settings    PEcAn settings object
+##' @param variables   variables to be extracted
+##' 
+##' @description Read Restart for SIPNET
+##' 
+##' @return X.vec      vector of forecasts
+##' @export
+##' 
+read.restart.SIPNET <- function(outdir,runid,time,settings,variables,sample_parameters=NULL){
 
-  forecast = IC
-  nens = nrow(IC)
-  ens = list()
+  if(sample_parameters == TRUE){
+    load(file.path(settings$outdir, "samples.Rdata"))
+    prior.sla <- mean(ensemble.samples[[which(names(ensemble.samples)!='soil')[1]]]$SLA) #HACK
+  }else{
+    load(file.path(settings$outdir, paste0("ensemble.samples.",settings$state.data.assimilation$prior,".Rdata")))
+    prior.sla <- ens.samples[[which(names(ensemble.samples)!='soil')[1]]]$SLA
+  }
+  
+  forecast <- list()
   unit.conv <-  0.001*2#  kgC/ha/yr to Mg/ha/yr
   
-  for(i in 1:nens){
-    ens[[i]] <- read.output(run.id[[i]],file.path(outdir, run.id[[i]]),
-                            start.year = time,end.year=time,
-                            variables=c("NPP","AbvGrndWood","TotSoilCarb","LeafC","SoilMoistFrac","SWE","Litter")
-    )
-    NPPm[i] <- mean(ens[[i]]$NPP)*unit.conv ## kg C m-2 s-1 -> Mg/ha/yr [Check]
-    last = length(ens[[i]]$NPP)
-    forecast$plantWood[i] = ens[[i]]$AbvGrndWood[last]*1000 ## kgC/m2 -> gC/m2
-    forecast$lai[i] = ens[[i]]$LeafC[last]*prior$SLA[i]*2 ## kgC/m2*m2/kg*2kg/kgC -> m2/m2
-    forecast$litter[i] = ens[[i]]$Litter[last]*1000 ##kgC/m2 -> gC/m2
-    forecast$soil[i] = ens[[i]]$TotSoilCarb[last]*1000 ## kgC/m2 -> gC/m2
-    forecast$litterWFrac[i] = ens[[i]]$SoilMoistFrac[last] ## unitless
-    forecast$soilWFrac[i] = ens[[i]]$SoilMoistFrac[last] ## unitless
-    forecast$snow[i] = ens[[i]]$SWE[last]*0.1 ## kg/m2 -> cm
-    #forecast$microbe[i] = NA
-  }
+  #Read ensemble output
+  ens <- read.output(runid = runid,outdir = file.path(outdir, runid),
+                     start.year = time, end.year=time,
+                     variables=variables)
+    
+    last = length(ens$NPP)
+  
+    forecast[[1]] <- mean(ens$NPP)*unit.conv ## kg C m-2 s-1 -> Mg/ha/yr [Check]
```

Comment:
if forecast is just holding scalars, why is it a list rather than a vector, since you just end up unlisting it a few lines later?